src/main/java/io/strimzi/kafka/bridge/http/HttpBridge.java

```diff
@@ -558,10 +608,14 @@ private void openapi(RoutingContext routingContext) {
     }
 
     private void metrics(RoutingContext routingContext) {
-        routingContext.response()
+        if (metricsCollector != null) {
+            routingContext.response()
                 .putHeader("Content-Type", "text/plain; version=0.0.4; charset=utf-8")
                 .setStatusCode(HttpResponseStatus.OK.code())
-                .end(metricsReporter.scrape());
+                .end(metricsCollector.scrape());
+        } else {
+            HttpUtils.sendResponse(routingContext, HttpResponseStatus.NOT_FOUND.code(), null, null);
```

Comment:
I saw you removed the 404 response from the openapi spec. I think we should keep it but just with a "description" field and no "content" section. Something similar happens for some 204 OK NO CONTENT responses.